packages/adapter-drizzle/src/lib/mysql.ts

```diff
@@ -100,15 +88,17 @@ export function MySqlDrizzleAdapter(
     schema
 
   return {
-    async createUser(data: Omit<AdapterUser, "id">) {
-      const id = randomUUID()
+    async createUser(data: AdapterUser) {
+      const hasDefaultId = getTableColumns(usersTable)["id"]["hasDefault"]
 
-      await client.insert(usersTable).values({ ...data, id })
+      await client
+        .insert(usersTable)
+        .values(hasDefaultId ? data : { ...data, id: crypto.randomUUID() })
 
       return client
         .select()
         .from(usersTable)
-        .where(eq(usersTable.id, id))
+        .where(eq(usersTable.email, data.email))
```

Comment:
there are no docs recommending putting an index on this column, so this lookup might be expensive? sucks that we don't have `insert().returning()`. can you run the defaultFn manually?

```ts
const id = usersTable._.$defaultFn() ?? crypto.randomUUID();
await client
  .insert(usersTable)
  .values({ ...data, id: crypto.randomUUID() })
```

wouldn't work with default's set on the db level though :/

Reply:
I have tested now and it works with `$defaultFn()` and `default()` (db level)

```ts
const defaultId = getTableColumns(usersTable)["id"]['default'] ?? getTableColumns(usersTable)["id"]['defaultFn']!()
```
but if user pass some function for generating id (e.g. `UUID()`)  then it won't work cause we can't get the result of this function before using in insert and in select statements. I also thought about selecting db default (e.g. `select (UUID())`) and assign it to the variable and then pass to all statements.

As for now, might be the best way to add information in docs about index on `email` field. What do you think about this?

Reply:
Sounds good to me